specifyweb/frontend/js_src/lib/components/WbPlanView/index.tsx

```diff
@@ -45,6 +48,48 @@ export function WbPlanViewWrapper(): JSX.Element | null {
   );
   useErrorContext('dataSet', dataSet);
 
+  const updateDataSetColumns = (lines: RA<MappingLine>) => {
```

Comment:
None of this should be needed

WbPlanView calls savePlan, which makes the network request. AND right after that it calls navigate to go back to the workbench. Then, workbench fetches the data set from the back-end, thus you don't need to update the dataSet state in this file manually OR to make a back-end request